Make aws/rds compatible with version 4 of AWS provider

vandrijevik opened the pull request:
When trying to upgrade Dickson’s Terraform installation to use version 4 of the AWS provider, we ran into a bunch of these errors:

```
╷
│ Error: Conflicting configuration arguments
│ 
│   with module.calclub-production-rds.aws_db_instance.mod,
│   on .terraform/modules/calclub-production-rds/aws/rds/main.tf line 66, in resource "aws_db_instance" "mod":
│   66:   username                     = var.username != "" ? var.username : "${var.name}${var.username_suffix}"
│ 
│ "username": conflicts with replicate_source_db
╵
```

Since we now have the module `aws/database_replica` for creating RDS replicas, this PR removes `var.source_db` from the module `aws/rds`, which fixes the problems and allows Terraform to proceed with a plan.

vandrijevik committed:
Remove source_db from aws/rds; for replicas, use aws/database_replica instead

pturley commented:
@vandrijevik is this backwards compatible with v3?

Also, I the big change in v4 is in S3 just FYI

vandrijevik committed:
Address Terraform warning about ignored attribute

The warning reads:

```
╷
│ Warning: Redundant ignore_changes element
│
│   on .terraform/modules/calclub-production-rds/aws/rds/main.tf line 39, in resource "aws_db_instance" "mod":
│   39: resource "aws_db_instance" "mod" {
│
│ Adding an attribute name to ignore_changes tells Terraform to ignore future changes to the argument in configuration after the object has been created, retaining the
│ value originally configured.
│
│ The attribute latest_restorable_time is decided by the provider alone and therefore there can be no configured value to compare with. Including this attribute in
│ ignore_changes has no effect. Remove the attribute from ignore_changes to quiet this warning.
│
│ (and 6 more similar warnings elsewhere)
╵
```

vandrijevik commented:
@pturley I am not sure, I didn’t test that since the only use of it is in `bookrags`, which is pointed at a particular release of this repo, so this PR won’t break that only reference. For any use cases where we would specify `source_db` in `aws/rds`, we should now be using `aws/database_replica`, anyway, I think? In any case, omitting an optional attribute (it’s the same attribute in v3 and it’s optional) is not likely to break v3 compatibility here.

I was able to get a functioning plan without major S3 updates, but there are some deprecation warnings around how we specify CORS configuration that need a bit more attention.

_Note:_ I just added https://github.com/tablexi/terraform_modules/pull/228/commits/8e47997c1bf49c0b4d49fa3d56cd155e6f137d17, and with that all of the RDS warnings and errors we had in Dickson have been addressed against AWS provider v4.
